designs/api-server-extra-args.md

```diff
@@ -0,0 +1,122 @@
+    apiServerExtraArgs:
+      "service-account-issuer": "https://{my-service-account-issuer-url}"
+      "service-account-jwks-uri": "https://{my-service-account-issuer-url}/openid/v1/jwks"
+      "service-account-signing-key-file": "/etc/kubernetes/pki/sa.key"
+      "service-account-key-file": "/etc/kubernetes/pki/sa.pub"
+    ...
+```
+
+**Validations:**
+
+* Validate that oidc flags are not configured in apiServerExtraArgs if OIDCConfig identity provider is already configured in the spec
+* Validate that the feature flag is enabled for configuring apiServerExtraArgs
+
+**Pros:**
+
+* Creates a standard way of exposing any flag for the control plane components
+* Gives more flexibility to the users in terms of validating the flag values for the api-server
+
+**Cons:**
+
+* Does not enforce OIDC compliance or any other validations on the allowed values for the flags
+
+## Alternate Solutions
+
+Allow users to configure the flags as a struct field in the cluster spec yaml
+
+**Schema:**
+
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Cluster
+metadata:
+  name: mgmt-cluster
+spec:
+  ...
+  controlPlaneConfiguration:
+    ...
+    [Proposal]
+    apiServerConfiguration:
+      # can add more flags here in the future
+      serviceAccountIssuer: 
+      - "https://{my-service-account-issuer-url}"
+      serviceAccountJwksUri: "https://{my-service-account-issuer-url}/openid/v1/jwks"
+    ...
+    [Alternate Option]
+    serviceAccountIssuerConfiguration:
+      issuer: "https://{my-service-account-issuer-url}"
+      jwksUri: "https://{my-service-account-issuer-url}/openid/v1/jwks"
+    ...
+```
+
+**Validations:**
+
+* Validate that the service-account-issuer flag is not configured for both controlPlaneConfiguration and PodIamConfig
+* Validate that the issuer url and jwks uri has the same domain
+* Validate that both the issuer url and jwks uri use https scheme as per the OIDC specification
+
+**Pros:**
+
+* Allows enforcing OIDC compliance for the service account flags of the api-server
+
+**Cons:**
+
+* Gives less flexibility to the users for configuring the flags in terms of number of validations
+* Does not provide a standard way to configure more flags in the future
+
+## Implementation Details:
+
+```
+apiServerExtraArgs:
+  "service-account-issuer": "https://{my-service-account-issuer-url}"
+  "service-account-jwks-uri": "https://{my-service-account-issuer-url}/openid/v1/jwks"
+```
+
+https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/control-plane-flags/#apiserver-flags  
+These flags will be fetched from the cluster spec and added to the apiServerExtraArgs in the ClusterConfiguration object during create and upgrade operations for generating control plane CAPI spec
+
+Users need to enable the feature flag `API_SERVER_EXTRA_ARGS_ENABLED=true` to configure the api server flags in the cluster spec. If it's not enabled, then it will throw an error when validating the cluster spec
+
```

Comment:
And we are doing this just to expose this functionality for now before we determine to support it officially with some more robust validations.